src/components/PeopplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,138 @@
+import { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { Link, useParams } from 'react-router-dom';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const { slug: selectedPersonSlug } = useParams();
+
+  useEffect(() => {
+    const getPeopleFromServer = async () => {
```

Comment:
I would move the function out `getPeopleFromServer` from useEffect and PeoplePage component and make it to return `peopleWithParents`. Then your useEffect would be small and easy to read like this:
```
  useEffect(() => {
    getPeopleFromServer().then(setPeople);
  }
```
But your variant is also valid. The only mistake it's that you didn't put `setPeople` as a dependency in the dependency array. React must have shown you a warning about that.

Reply:
Also, you need to have error handling for all requests. Open your demo on the home page, hard refresh, open dev tools, set the network throttling as Offline, and click on the People page. You'll get an error and your users will see endless loading, which is super confusing for them. It's good to show some message explaining what exactly happened.

Reply:
Try to add one more state variable using a state that will show if a request is loading or no. Every request usually goes with 3 states:
- loader state
- data state
- error state